tests/serve/common.py

```
@@ -185,14 +186,26 @@ def run_serve_deployment(
-                    response = send_request(
-                        url=payload.url(),
-                        payload=payload.body,
-                        timeout=payload.timeout,
-                        method=payload.method,
-                        stream=payload.http_stream,
+                    # Re-issue the request (server stays up) on validation
+                    # failure when payload.retries > 0. See tests/README.md
+                    # "Flaky Tests" for when this is appropriate.
+                    def _send_and_check() -> None:
+                        response = send_request(
+                            url=payload.url(),
```

Ruff (0.15.12)

[warning] Function definition does not bind loop variable `payload`

(B023)